py-polars/tests/unit/datatypes/test_enum.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+
+import polars as pl
+from polars import StringCache
+from polars.testing import assert_series_equal
+
+
+def test_enum_creation() -> None:
+    s = pl.Series([None, "a", "b"], dtype=pl.Enum(categories=["a", "b"]))
+    assert s.null_count() == 1
+    assert s.len() == 3
+    assert s.dtype == pl.Enum(categories=["a", "b"])
+
+
+def test_enum_non_existent() -> None:
+    with pytest.raises(
+        pl.OutOfBoundsError,
```

Comment:
I'm not sure `OutOfBounds` is the correct characterisation here. I guess it should be `ComputeError` or ideally something more specific, e.g. `InvalidCategoryError` or something.